fix(mqtt-bridge): add docs (#1311)
emqtt/emq-docs-cn: en_US/data-integration/data-bridge-mqtt.md

# MQTT Bridge

> The MQTT bridge is a channel for EMQX to communicate with other MQTT services, either EMQX or other services that support the MQTT protocol. The MQTT bridge can either subscribe from external services or publish messages to external services. However, it only supports a one-way mode of operation. Can only be a producer, or a consumer. If a bidirectional bridge is required, multiple MQTT bridges with different directions need to be created to complete the bidirectional flow of data.

Reviewer: ```suggestion
The MQTT bridge is a channel for EMQX to communicate with other MQTT services, either EMQX or other services that support the MQTT protocol. The MQTT bridge can either subscribe from the external services or publish messages to the external services. However, it only supports unidirectional communication: it can only be a producer, or a consumer. If a bidirectional bridge is required, multiple MQTT bridges with different directions need to be created to complete the bidirectional flow of data.
```

> The MQTT bridge is a channel for EMQX to communicate with other MQTT services, either EMQX or other services that support the MQTT protocol. The MQTT bridge can either subscribe from external services or publish messages to external services. However, it only supports a one-way mode of operation. Can only be a producer, or a consumer. If a bidirectional bridge is required, multiple MQTT bridges with different directions need to be created to complete the bidirectional flow of data.

Reviewer 2: ```suggestion
The MQTT bridge is a channel for EMQX to communicate with other MQTT services, either EMQX or other services that support the MQTT protocol. The MQTT bridge can either subscribe from external services or publish messages to external services. However, a single bridge only supports a one-way mode of operation: it can be either a producer or a consumer, but not both. If a bidirectional bridge is required, multiple MQTT bridges with different directions need to be created to complete the bidirectional flow of data.
```

| connector | MQTT connector | connector() |  yes | MQTT Connecter configurations |

### List of `egress` type MQTT Bridge configurations

Bridges local messages to external services.

| Parameter Name | Description | Type | Required | Range |
| -- | -- | -- | -- | -- |
| enable       | Enable or disable | Boolean |  yes | - |
| direction    | Bridging direction. </br>`ingress` means subscribe messages from external service, publish to local</br>`egress` means publish messages from local to external service | String | yes | ingress |
| remote_topic | The topic published to external service        | String      | yes | -           |
| remote_qos   | The QoS published to external service          | Integer     | yes | 0 \| 1 \| 2 |
| retain       | The Retain flag published to external service  | Boolean     | yes | -           |
| payload      | The Payload flag published to external service | String      | yes | -           |
| local_topic  | Local Topic for getting data                   | String      | yes | -           |
| connector    | MQTT Connector                                 | connector() | yes | MQTT Connecter configurations |

> ### MQTT Connecter configurations

Reviewer 2: ```suggestion
### MQTT Connector configurations
```

      direction = "egress"
      enable = true
      local_topic = "local/topic/egress"
      remote_qos = 0
      remote_topic = "remote/topic/egress"
      payload = "${payload}"
      retain = false
    }
  }
}
```

Open EMQX Dashboard and click `Data Integration` - `Data Bridges` on the right side to see the two bridges created.
![image](./assets/rules/mqtt_bridge/dashboard_show_bridges.png)

::: tip
Note that the 5.0.0 Dashboard does not support detachment rules at this time, using the MQTT Bridge alone, which can only be created using the configuration file
:::

### The `ingress bridge` message flow

```txt
 +-------------------------+
 | Remote                  |          +--------+
 | EMQX Broker             |<---------| Client |
 |                         |          +--------+
 +-------------------------+
             |
             |
             V
  +----------------------+
  |  MQTT Bridge Ingress |
  +----------------------+
             |
             V
  +------------------------+
  | Local                  |          +--------+
  | EMQX Broker            |--------->| Client |
  |                        |          +--------+
  +------------------------+

```

> Using the desktop MQTT client MQTTX, create two connections, one to the local and one to the external service. And subscribe to `local/topic/ingress` locally.

Reviewer: ```suggestion
Using the desktop MQTT client MQTTX, create two connections: one to the local and one to the external service. Then subscribe to `local/topic/ingress` locally.
```
Also regarding the last sentence, it's not very clear what does it mean "subscribe locally"

![image](./assets/rules/mqtt_bridge/local_sub.png)

Publish a message to an external service with a Topic of `remote/topic/ingress`.

![image](./assets/rules/mqtt_bridge/remote_pub.png)

Check the local connection, the message has been bridged locally by the MQTT Bridge.

![image](./assets/rules/mqtt_bridge/local_recv.png)

### The `egress bridge` message flow

```txt
 +-------------------------+
 | Remote                  |          +--------+
 | EMQX Broker             |--------->| Client |
 |                         |          +--------+
 +-------------------------+
             ^
             |
             |
  +----------------------+
  |  MQTT Bridge Ingress |
  +----------------------+
             ^
             |
  +------------------------+
  | Local                  |          +--------+
  | EMQX Broker            |<---------| Client |
  |                        |          +--------+
  +------------------------+

```

> Using the desktop MQTT client MQTTX, create two connections, one to the local and one to the external service. And subscribe to `remote/topic/egress` on the external service.

Reviewer: ```suggestion
Using the desktop MQTT client MQTTX, create two connections, one to the local and one to the external service. Then subscribe to `remote/topic/egress` on the external service.
```
